vendor/cloud.google.com/go/auth/credentials/filetypes.go

```diff
@@ -36,6 +36,8 @@ func fileCredentials(b []byte, opts *DetectOptions) (*auth.Credentials, error) {
 	var projectID, universeDomain string
 	var tp auth.TokenProvider
 	switch fileType {
+	case credsfile.UnknownCredType:
+		return nil, errors.New("credentials: unsupported unidentified file type")
```

Comment:
This is also requires https://github.com/GoogleContainerTools/kpt-config-sync/pull/1833